### PR TITLE
Improve resource docs

### PR DESCRIPTION
**Pull request by leplatrem**

### CODE REVIEW

File: cliquet/resource.py [hunk 1]
```diff
@@ -83,7 +96,8 @@ def schema(self):
         return CorniceSchema.from_colander(colander_schema)
 
     def is_known_field(self, field):
-        """Return the True if the field is defined in the resource mapping.
+        """Return the `True` if the field is defined in the resource mapping.
```
Comment (Natim) [inline cliquet/resource.py]: Return ``True`` if the field is defined in the resource mapping.

File: cliquet/resource.py [hunk 2]
```diff
@@ -102,7 +116,8 @@ def is_known_field(self, field):
         cors_headers=('Next-Page', 'Total-Records', 'Last-Modified')
     )
     def collection_get(self):
-        """Collection `GET` endpoint."""
+        """Collection `GET` endpoint. See :meth:`cliquet.resource.get_records`.
```
Comment (Natim) [inline cliquet/resource.py]: ``GET``

File: cliquet/resource.py [hunk 3]
```diff
@@ -122,7 +137,9 @@ def collection_get(self):
 
     @resource.view(permission='readwrite')
     def collection_post(self):
-        """Collection `POST` endpoint."""
+        """Collection `POST` endpoint.
```
Comment (Natim) [inline cliquet/resource.py]: ``POST``

**Comment by Natim** [comment]
Except for the

    ``

Everything looks ok.

File: cliquet/cache/__init__.py [hunk 4]
```diff
@@ -39,51 +39,44 @@ def ping(self):
...
+        :param str key: key
         :returns: the stored value or None if missing.
-        :rtype: string
+        :rtype: str
         """
         raise NotImplementedError
 
     def delete(self, key):
-        """Delete the value of the specified key.
+        """Delete the value of the specified `key`.
```
Comment (Natim) [inline cliquet/cache/__init__.py]: ``key``

File: cliquet/cache/__init__.py [hunk 5]
```diff
@@ -39,51 +39,44 @@ def ping(self):
...
+
+        :param str key: key
+        :param str value: value to store
+        :param float ttl: expire after number of seconds
         """
         raise NotImplementedError
 
     def get(self, key):
-        """Obtain the value of the specified key.
+        """Obtain the value of the specified `key`.
```
Comment (Natim) [inline cliquet/cache/__init__.py]: ``key``

File: cliquet/cache/__init__.py [hunk 6]
```diff
@@ -39,51 +39,44 @@ def ping(self):
...
-        :type key: string
+        :param str key: key
         :returns: number of seconds or negative if no TTL.
         :rtype: float
         """
         raise NotImplementedError
 
     def expire(self, key, ttl):
-        """Set the expiration value of the specified key.
+        """Set the expiration value `ttl` for the specified `key`.
```
Comment (Natim) [inline cliquet/cache/__init__.py]: ``ttl`` ...  ``key``

File: cliquet/cache/__init__.py [hunk 7]
```diff
@@ -26,8 +26,8 @@ def flush(self):
     def ping(self):
         """Test that cache backend is operationnal.
 
-        :returns: `True` is everything is ok, `False` otherwise.
-        :rtype: boolean
+        :returns: ``True`` is everything is ok, ``False`` otherwise.
```
Comment (Natim) [inline cliquet/cache/__init__.py]: ``True`` ... ``False``

File: README.rst [hunk 8]
```diff
@@ -16,7 +16,7 @@ Cliquet
     :target: https://coveralls.io/r/mozilla-services/cliquet
 
 
-Cliquet is a `Cornice <https://cornice.rtfd.org>`_ application to ease
+*Cliquet* is a `Cornice <http://cornice.readthedocs.org>`_ application to ease
```
Comment (Natim) [inline README.rst]: https

File: cliquet/cache/__init__.py [hunk 9]
```diff
@@ -39,51 +39,44 @@ def ping(self):
...
     def set(self, key, value, ttl=None):
-        """Store a value with the specified key.
-
-        :param key: key
-        :type key: string
-        :param value: value to store
-        :type value: string
-        :param ttl: expire after number of seconds
-        :type ttl: float
+        """Store a value with the specified `key`. If `ttl` is provided,
```
Comment (Natim) [inline cliquet/cache/__init__.py]: ``ttl``

File: cliquet/resource.py [hunk 10]
```diff
@@ -73,7 +86,11 @@ def name(self):
 
     @property
     def schema(self):
-        """Resource schema, depending on HTTP verb."""
+        """Resource schema, depending on HTTP verb.
+
+        :returns: a :class:`~cornice:cornice.schemas.CorniceSchema` object
+            built from this resource :attr:`mapping <.BaseResource.mapping>`.
```
Comment (Natim) [inline cliquet/resource.py]: is it starting with a ``.`` on purpose? : ``mapping <.BaseResource.mapping>``

File: cliquet/resource.py [hunk 11]
```diff
@@ -83,10 +100,10 @@ def schema(self):
         return CorniceSchema.from_colander(colander_schema)
 
     def is_known_field(self, field):
-        """Return the True if the field is defined in the resource mapping.
-        :param field: Field name
-        :type field: string
-        :rtype: boolean
+        """Return ``True`` if `field` is defined in the resource mapping.
```
Comment (Natim) [inline cliquet/resource.py]: ``field``

File: cliquet/resource.py [hunk 12]
```diff
@@ -402,8 +524,11 @@ def apply_changes(self, record, changes):
                     changes.pop('position', None)
                 return super(MyResource, self).apply_changes(record, changes)
 
-        :raises: :class:`pyramid.httpexceptions.HTTPBadRequest` if result does
-            not comply with resource schema.
+        :raises: :exc:`~pyramid:pyramid.httpexceptions.HTTPBadRequest`
+            if result does not comply with resource schema.
+
+        :returns: the new record with `changes` applied.
```
Comment (Natim) [inline cliquet/resource.py]: ``changes``

File: cliquet/schema.py [hunk 13]
```diff
@@ -54,20 +61,37 @@ class Options:
...
+                    unique_fields = ('reference',)
         """
-        readonly_fields = ('id', 'last_modified')
-        """Fields that cannot be updated"""
-
         unique_fields = ('id', 'last_modified')
-        """Fields that must have unique values for the user collection"""
+        """Fields that must have unique values for the user collection.
+        During records creation and modification, a conflict error will be
+        raised if unicity is about to be violated."""
```
Comment (Natim) [inline cliquet/schema.py]: ``"""`` should be on the line below.

File: cliquet/storage/__init__.py [hunk 14]
```diff
@@ -61,40 +61,39 @@ def ping(self, request):
             return False
 
     def collection_timestamp(self, resource, user_id):
-        """Get the highest timestamp of every records in this resource for
-        this user.
+        """Get the highest timestamp of every records in this `resource` for
```
Comment (Natim) [inline cliquet/storage/__init__.py]: ``resource``

File: cliquet/storage/__init__.py [hunk 15]
```diff
@@ -61,40 +61,39 @@ def ping(self, request):
...
+        this `user_id`.
```
Comment (Natim) [inline cliquet/storage/__init__.py]: ``user_id``

File: cliquet/storage/__init__.py [hunk 16]
```diff
@@ -61,40 +61,39 @@ def ping(self, request):
...
         :returns: the latest timestamp of the collection.
-        :rtype: integer
+        :rtype: int
         """
         raise NotImplementedError
 
     def create(self, resource, user_id, record):
-        """Create the specified record in this resource for this user.
-        Assign the id to the record, using the `resource.id_field` attribute.
+        """Create the specified `record` in this `resource` for this `user_id`.
```
Comment (Natim) [inline cliquet/storage/__init__.py]: ``resource`` ... ``user_id``

File: cliquet/storage/__init__.py [hunk 17]
```diff
@@ -102,18 +101,16 @@ def create(self, resource, user_id, record):
         raise NotImplementedError
 
     def get(self, resource, user_id, record_id):
-        """Retrieve the record with specified id, or raise error if not found.
+        """Retrieve the record with specified `record_id`, or raise error
```
Comment (Natim) [inline cliquet/storage/__init__.py]: ``record_id``

File: cliquet/storage/__init__.py [hunk 18]
```diff
@@ -121,27 +118,23 @@ def get(self, resource, user_id, record_id):
         raise NotImplementedError
 
     def update(self, resource, user_id, record_id, record):
-        """Overwrite the record with the specified id.
+        """Overwrite the `record` with the specified `record_id`.
```
Comment (Natim) [inline cliquet/storage/__init__.py]: ``record`` ... ``record_id``

File: cliquet/storage/__init__.py [hunk 19]
```diff
@@ -149,25 +142,24 @@ def update(self, resource, user_id, record_id, record):
         raise NotImplementedError
 
     def delete(self, resource, user_id, record_id):
-        """Delete the record with specified id, and raise error if not found.
+        """Delete the record with specified `record_id`, and raise error
```
Comment (Natim) [inline cliquet/storage/__init__.py]: ``record_id``

File: cliquet/storage/__init__.py [hunk 20]
```diff
@@ -175,16 +167,15 @@ def delete(self, resource, user_id, record_id):
         raise NotImplementedError
 
     def delete_all(self, resource, user_id, filters=None):
-        """Delete all records in this resource for this user.
+        """Delete all records in this `resource` for this `user_id`.
```
Comment (Natim) [inline cliquet/storage/__init__.py]: ``resource`` ... ``user_id``

File: cliquet/storage/__init__.py [hunk 21]
```diff
@@ -193,38 +184,36 @@ def delete_all(self, resource, user_id, filters=None):
 
     def get_all(self, resource, user_id, filters=None, sorting=None,
                 pagination_rules=None, limit=None, include_deleted=False):
-        """Retrieve all records in this resource for this user.
+        """Retrieve all records in this `resource` for this `user_id`.
```
Comment (Natim) [inline cliquet/storage/__init__.py]: ``resource`` ... ``user_id``

File: docs/resource.rst [hunk 22]
```diff
@@ -1,7 +1,78 @@
...
+
+*Cliquet* provides a basic component to build resource oriented APIs.
+In most cases, the main customization consists in defining the schema of the
+records for this resource.
+
+
+Full example
+============
+
+.. code-block :: python
```
Comment (Natim) [inline docs/resource.rst]: .. code-block:: python

File: docs/resource.rst [hunk 23]
```diff
@@ -1,7 +1,78 @@
...
+
+
+.. _resource-schema:
+
+Resource Schema
+===============
+
+Override the base schema to add extra fields using the `Colander API <http://docs.pylonsproject.org/projects/colander/>`_.
+
+.. code-block :: python
```
Comment (Natim) [inline docs/resource.rst]: .. code-block:: python

File: cliquet/resource.py (hunk 10 above)
Comment (leplatrem) [inline cliquet/resource.py]: yes it means current module

**Pull request merged**